lib/stimulus_reflex/element.rb

```diff
@@ -29,6 +29,10 @@ def dataset
     @dataset ||= OpenStruct.new(build_underscored(data_attrs))
   end
 
+  def to_dom_id
+    "##{id}"
```

Comment:
So where does `id` come from?

And what happens if `id` isn't present? Do all ViewComponents have an `id` attribute?

Reply:
Good point, I forgot a `raise` here. also, you're right, it's an accessor on `element`, I'm just wondering where it comes from myself 🤔

Reply:
actually I think it's `super build_undescored` in `StimulusReflex::Element#initialize`

Reply:
I am not aware of any reason `element` would have an `id`, given that it's not declared in this class.

Reply:
here: https://github.com/stimulusreflex/stimulus_reflex/blob/master/lib/stimulus_reflex/reflex_data.rb#L37

and here: https://github.com/stimulusreflex/stimulus_reflex/blob/master/lib/stimulus_reflex/element.rb#L12

It works because it's an OpenStruct (I think)

Reply:
You are right!

So what are you thinking? Custom exception class like `NoIDException` or the like?

Reply:
Yep, I will add that!